policyengine_us/variables/gov/states/nj/tax/income/exclusions/nj_other_retirement_income_exclusion.py

```diff
@@ -0,0 +1,100 @@
+        p = parameters(period).gov.states.nj.tax.income.exclusions.retirement
+
+        # Get the total income (line 27) for head and spouse above age threshold.
+        person = tax_unit.members
+        is_head = person("is_tax_unit_head", period)
+        is_spouse = person("is_tax_unit_spouse", period)
+        filing_status = tax_unit("filing_status", period)
+        status = filing_status.possible_values
+        joint = filing_status == status.JOINT
+        age_eligible = person("age", period) >= p.age_threshold
+        eligible_head = age_eligible * is_head
+        eligible_spouse = age_eligible * is_spouse
+        gross_income = person("irs_gross_income", period)
+        exempt_interest_income = person("tax_exempt_interest_income", period)
+        exempt_pension_income = person("tax_exempt_pension_income", period)
+        fed_taxable_ss = person("taxable_social_security", period)
+        person_qualifying_income = (
+            gross_income
+            - exempt_interest_income
+            - exempt_pension_income
+            - fed_taxable_ss
+        )
+        qualifying_income = where(
+            joint,
+            tax_unit.sum(
+                person_qualifying_income * (eligible_head + eligible_spouse)
+            ),
+            tax_unit.sum(person_qualifying_income * eligible_head),
+        )
+
+        # Determine income eligibility.
+        income_eligible = (
+            qualifying_income <= p.other_retirement_income_threshold
+        )
+
+        # Get the exclusion percentage based on filing status and income.
+        # Same logic as pension/retirement exclusion.
+        filing_status = tax_unit("filing_status", period)
+        status = filing_status.possible_values
+        exclusion_percentage = select(
+            [
+                filing_status == status.SINGLE,
+                filing_status == status.JOINT,
+                filing_status == status.HEAD_OF_HOUSEHOLD,
+                filing_status == status.WIDOW,
+                filing_status == status.SEPARATE,
+            ],
+            [
+                p.percentage.single.calc(qualifying_income),
+                p.percentage.joint.calc(qualifying_income),
+                p.percentage.head_of_household.calc(qualifying_income),
+                p.percentage.widow.calc(qualifying_income),
+                p.percentage.separate.calc(qualifying_income),
+            ],
+        )
+
+        # Get the potential exclusion amount for the tax unit.
+        # If all the income in qualifying_income were eligible for exclusion.
+        max_exclusion = qualifying_income * exclusion_percentage
+
+        # Get the tax unit's pension/retirement exclusion from line 28a.
+        pension_retirement_exclusion = tax_unit(
+            "nj_pension_retirement_exclusion", period
+        )
+
+        # Get the excess (not excluded in potential/retirement) exclusion amount.
+        excess = max_exclusion - pension_retirement_exclusion
```

Comment:
Then you can skip the `positive_excess` below
```suggestion
        excess = max_(0, max_exclusion - pension_retirement_exclusion)
```